packages/create-turbo/src/commands/create/createProject.ts

```diff
@@ -0,0 +1,231 @@
+import retry from "async-retry";
+import chalk from "chalk";
+import fs from "fs-extra";
+import path from "path";
+import semverPrerelease from "semver/functions/prerelease";
+
+import {
+  downloadAndExtractExample,
+  downloadAndExtractRepo,
+  getRepoInfo,
+  existsInRepo,
+  hasRepo,
+  RepoInfo,
+} from "../../utils/examples";
+import { addGitIgnore } from "../../utils/git";
+import { isFolderEmpty } from "../../utils/isFolderEmpty";
+import { isWriteable } from "../../utils/isWriteable";
+import { turboLoader, error } from "../../logger";
+import cliPkgJson from "../../../package.json";
+
+export class DownloadError extends Error {}
+
+export async function createProject({
+  appPath,
+  projectName,
+  example,
+  examplePath,
+}: {
+  appPath: string;
+  projectName: string;
+  example: string;
+  examplePath?: string;
+}): Promise<{
+  cdPath: string;
+  hasPackageJson: boolean;
+  availableScripts: Array<string>;
+}> {
+  let repoInfo: RepoInfo | undefined;
+  let repoUrl: URL | undefined;
+  const isDefaultExample = example === "basic" || example === "default";
+
+  try {
+    repoUrl = new URL(example);
+  } catch (err: any) {
+    if (err.code !== "ERR_INVALID_URL") {
+      error(err);
+      process.exit(1);
+    }
+  }
+
+  if (repoUrl) {
+    if (repoUrl.origin !== "https://github.com") {
```

Comment:
I'm wondering if we'll end up with anyone who omits the scheme...?

Reply:
Hmm good call, we could support both